### DISCUSSION TITLE
Some confusion about storing `HTMLElement` in state

### DISCUSSION THREAD

**Discussion by lmk123** [pmndrs/valtio#1020] [Q&A] [answered]
https://codesandbox.io/p/sandbox/valtio-to-do-list-forked-9l3mf6

I use `createRoot().render()` to manually manage the creation and destruction of React components. Now, I need to pass an `HTMLElement` to my component. So, I store the `HTMLElement` in the state, then `subscribe` for changes to the state and re-render the component using `createRoot().render()` whenever there is a change.

However, in practice, I found that even if I just read the HTMLElement's attributes it would report an error. Tried using `snapshot` but that didn't help.

So my question is: can't I store the HTMLElement in the state, even if I just read some of its properties and don't make any changes?

> [ref](https://valtio.dev/docs/api/advanced/ref) doesn't seem to work in this case either, since it doesn't allow passing `null` as an argument.

II'd love to automate and manage this process with Valito. If I can't use Valito I'll have to re-render manually:

```js
const state = { a: null, b: 2 }

state.a = document.createElement('div')
// ... some other change ...

// manually render
root.render(<App ele={state.a} count={state.b} />)
```

**Comment by dai-shi** [comment, marked as answer]
How about this?
```js
state.a = ref(document.createElement('div'))
```

**Reply by lmk123** [reply]
So simple...

When I read the documentation, I thought `ref` could only be used in `proxy()` :joy:

**Reply by dai-shi** [reply]
Yeah, fyi, actually, ref is just making a side effect, so it just returns the same object:
https://github.com/pmndrs/valtio/blob/24753f6f64cfd256a21055f595d2bd772b3149bb/src/vanilla.ts#L319-L322
